File: php/src/Services/Strategies/AgedBrieStrategy.php
<?php

namespace GildedRose\Services\Strategies;

use GildedRose\Item;

class AgedBrieStrategy extends DefaultStrategy
{
    public function setSellIn(): Item
    {
        $this->item->sell_in--;
        return $this->item;
    }

    public function setQuality(): Item
    {
        $this->item->quality = ($this->item->quality + $this->degradeRate) > self::MAX_QUALITY_VALUE ?
            self::MAX_QUALITY_VALUE : $this->item->quality += $this->degradeRate;
        return $this->item;
    }
}

File: csharp.NUnit/GildedRose/Strategies/AgedBrieStrategy.cs
namespace GildedRoseKata.Strategies;

public class AgedBrieStrategy : BaseUpdateStrategy
{
    public override void UpdateQuality(Item item)
    {
        IncreaseQuality(item);
        DecreaseSellIn(item);

        if (item.SellIn < 0)
        {
            IncreaseQuality(item);
        }
    }
}
